packages/hoprd/crates/hopr-lib/src/lib.rs

```diff
@@ -206,10 +206,72 @@ mod native {
             self.aliases.read().await.clone()
         }
 
+        pub async fn get_balance(&self, balance_type: BalanceType) -> errors::Result<Balance> {
+            self.chain_api
+                .rpc()
+                .get_balance(self.me_onchain(), balance_type)
+                .await
+                .map_err(|e| HoprLibError::ChainApi(e.into()))
+        }
+
+        pub async fn get_safe_balance(&self, balance_type: BalanceType) -> errors::Result<Balance> {
+            self.chain_api
+                .rpc()
+                .get_balance(self.safe_module_cfg.safe_address, balance_type)
+                .await
+                .map_err(|e| HoprLibError::ChainApi(e.into()))
+        }
+
+        pub fn get_safe_config(&self) -> SafeModule {
+            self.safe_module_cfg.clone()
+        }
+
+        pub async fn get_channel_closure_notice_period(&self) -> errors::Result<Duration> {
+            self.chain_api
+                .rpc()
+                .get_channel_closure_notice_period()
+                .await
+                .map_err(|e| HoprLibError::ChainApi(e.into()))
+        }
+
         pub fn chain_config(&self) -> ChainNetworkConfig {
             self.chain_cfg.clone()
         }
 
+        // Move this single-purpose function to core-ethereum-api ?
+        async fn wait_for_funds(&self) -> errors::Result<()> {
+            let max_delay = Duration::from_secs(200);
+            let multiplier = 1.05;
+            let min_native_balance = Balance::new_from_str(MIN_NATIVE_BALANCE, BalanceType::Native);
+
+            let mut current_delay = Duration::from_secs(2);
+
+            while current_delay <= max_delay {
+                match self
+                    .chain_api
+                    .rpc()
+                    .get_balance(self.me_onchain(), BalanceType::Native)
+                    .await
+                {
+                    Ok(current_balance) => {
+                        info!("current balance is {}", current_balance.to_formatted_string());
+                        if current_balance.gte(&min_native_balance) {
+                            info!("node is funded");
+                            return Ok(());
+                        } else {
+                            warn!("still unfunded, trying again soon");
+                        }
+                    }
+                    Err(e) => error!("error while querying for balance: {e}"),
```

Comment:
the error part will be in the log output, maybe let's make this more log reader friendly:
```suggestion
                    Err(e) => error!("failed to fetch balance from chain: {e}"),
```